crates/tap-agent/src/agent/sender_allocation.rs

```diff
@@ -1262,19 +1269,149 @@ pub mod tests {
         handle.stopped().await;
     }
 
+    async fn execute<Fut>(
+        pgpool: PgPool,
+        populate: impl FnOnce(PgPool) -> Fut,
+        amount_of_receipts: u64,
```

Comment:
Could you invert the order of arguments so we have a Closure as the last one? It's hard to find the last argument.

```suggestion
        pgpool: PgPool,
        amount_of_receipts: u64,
        populate: impl FnOnce(PgPool) -> Fut,
```